src/Native/Unix/System.Security.Cryptography.Native/openssl.c

```diff
@@ -336,7 +335,7 @@ extern "C" int32_t CryptoNative_GetAsn1StringBytes(ASN1_STRING* asn1, uint8_t* p
         return -length;
     }
 
-    memcpy_s(pBuf, UnsignedCast(cBuf), asn1->data, UnsignedCast(length));
+    memcpy_s(pBuf, (size_t)cBuf, asn1->data, (size_t)length);
```

Comment:
The UnsignedCast contains an assert that the value we are casting is >= 0. With this change, we are losing that. It seems we should rather define a C version of the UnsignedCast to keep that debugging behavior.

Reply:
Makes sense. Something that hasn't been taken care of in earlier PRs of same sorts. I can do it now or I can do it in the next to the next PR, that I mentioned to @bartonjs https://github.com/dotnet/corefx/pull/30376#discussion_r195300339. Whichever you prefer :)

Reply:
Do you have a recommendation how should the C version of UnsignedCast and SignedCast looks like?

Reply:
Just assert that the input is in the right range and return the cast value. You'll obviously need to create couple of variants based on the types used in the casts.

Reply:
Thanks, i have pushed a commit.